mango-python/bdgenomics/mango/QC.py

```diff
@@ -113,4 +114,85 @@ def plot(self, normalize = False, cumulative = False, xScaleLog = False, yScaleL
             plt.legend(loc=2, shadow = True, bbox_to_anchor=(1.05, 1))
             plt.show()
 
-        return coverageDistributions
\ No newline at end of file
+        return coverageDistributions
+
+## Plots alignment distribution for AlignmentRDD using the cigar string
+class AlignmentDistribution(object):
+    """
+    QC provides preprocessing functions for visualization
+    of various quality control.
+    """
+
+    def __init__(self, alignmentRDD):
+        """
+        Initializes a AlignmentDistribution class.
+        Computes the coverage distribution of multiple coverageRDDs.
+        :param bdgenomics.adam.rdd.AlignmentRDD alignmentRDD: A single alignment RDD
+        """
+        cigars = alignmentRDD.toDF().rdd.map(lambda r: r["cigar"])
+        self.alignments = cigars.flatMap(lambda r: Cigar(r).items()) \
+                    .map(lambda a: ((a[1],a[0]),1)) \
+                    .filter(lambda a: a[0][0] is not None) \
+                    .reduceByKey(lambda x,y:x+y) \
+                    .map(lambda r:(r[0][0], Counter({r[0][1]:r[1]}))) \
+                    .reduceByKey(lambda x,y:x+y) \
+                    .collect()
```

Comment:
What if the alignment file is very large? will collecting all of the cigar items be a memory issue on the driver?

Reply:
As long as the user specifies a reasonable bin size (i.e. don't materialize >millions of bins), there shouldn't be an issue on the driver since all of the data is already aggregated by bin.